server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -2228,9 +2228,6 @@ public ActionRequestValidationException validate(
             if (sorts() != null) {
                 specified.add(SORT_FIELD.getPreferredName());
             }
-            if (rescores() != null) {
-                specified.add(RESCORE_FIELD.getPreferredName());
-            }
```

Comment:
Seems like this is still technically not supported. No actual user provided retrievers actually support rescore. 

Is it really ok to remove this validation? I realize you did this for a test, but we need to ensure `main` is always deployable in whatever state its in.